internal/resolution/variablesources/installed_package.go

```diff
@@ -26,18 +29,20 @@ func (r *InstalledPackageVariableSource) GetVariables(ctx context.Context) ([]de
 		return nil, err
 	}
 
+	vr, err := mmsemver.NewConstraint(r.bundleVersion)
+	if err != nil {
+		return nil, err
+	}
+
 	// find corresponding bundle for the installed content
-	resultSet := catalogfilter.Filter(allBundles, catalogfilter.WithBundleImage(r.bundleImage))
+	resultSet := catalogfilter.Filter(allBundles, catalogfilter.And(
+		catalogfilter.WithPackageName(r.pkgName),
+		catalogfilter.WithName(r.bundleName),
+		catalogfilter.InMastermindsSemverRange(vr),
+	))
 	if len(resultSet) == 0 {
 		return nil, r.notFoundError()
 	}
-
-	// TODO: fast follow - we should check whether we are already supporting the channel attribute in the operator spec.
-	//       if so, we should take the value from spec of the operator CR in the owner ref of the bundle deployment.
-	//       If that channel is set, we need to update the filter above to filter by channel as well.
```

Comment:
I believe this is no longer relevant given that we filter by package name, bundle name and bundle version.